lib/awful/screenshot.lua

```diff
@@ -0,0 +1,870 @@
+  args.method = "screen"
+  return module(args)
+end
+
+--- Take client window screenshots.
+--
+-- This is a wrapper constructor for a client window screenshot. See the main
+-- constructor, new(), for details about the arguments.
+--
+-- @constructorfct awful.screenshot.client
+-- @tparam[opt] table args Table of arguments to pass to the constructor
+-- @treturn screenshot The screenshot object
+function module.client(args)
+  -- Looking at the properties and functions available, I'm not sure it is
+  -- wise to allow a "target" client argument, but if we want to add it as
+  -- arg 3 (which will match the screen ordering), we can.
+  local args = (type(args) == "table" and args) or {}
+  args.method = "client"
+  return module(args)
+end
+
+--- Launch an interactive snipper tool to take cropped shots.
+--
+-- This is a wrapper constructor for a snipper tool screenshot. See the main
+-- constructor, new(), for details about the arguments.
+--
+-- @constructorfct awful.screenshot.snipper
+-- @tparam[opt] table args Table of arguments to pass to the constructor
+-- @treturn screenshot The screenshot object
+function module.snipper(args)
+  local args = (type(args) == "table" and args) or {}
+  args.method = "snipper"
+  return module(args)
+end
+
+--- Take a cropped screenshot of a defined geometry.
+--
+-- This is a wrapper constructor for a snip screenshot (defined geometry). See
+-- the main constructor, new(), for details about the arguments.
+--
+-- @constructorfct awful.screenshot.snip
+-- @tparam[opt] table args Table of arguments to pass to the constructor
+-- @treturn screenshot The screenshot object
+function module.snip(args)
+  local args = (type(args) == "table" and args) or {}
+  args.method = "snip"
+  return module(args)
+end
+
+-- Various accessors for the screenshot object returned by any public
+-- module method.
+
+--- Get screenshot directory property.
+--
+-- @property directory
+function screenshot:get_directory()
+  return self._private.directory
+end
+
+--- Set screenshot directory property.
+--
+-- @tparam string directory The path to the screenshot directory
+function screenshot:set_directory(directory)
+  if type(directory) == "string" then
+    local checked_dir = check_directory(directory)
+    if checked_dir then
+      self._private.directory = checked_dir
+    end
+  end
+end
+
+--- Get screenshot prefix property.
+--
+-- @property prefix
+function screenshot:get_prefix()
+  return self._private.prefix
+end
+
+--- Set screenshot prefix property.
+--
+-- @tparam string prefix The prefix prepended to screenshot files names.
+function screenshot:set_prefix(prefix)
+  if type(prefix) == "string" then
+    local checked_prefix = check_prefix(prefix)
+    if checked_prefix then
+      self._private.prefix = checked_prefix
+    end
+  end
+end
+
+--- Get screenshot filepath.
+--
+-- @property filepath
+function screenshot:get_filepath()
+  return self._private.filepath
+end
+
+--- Set screenshot filepath.
```

Comment:
If you have a `@property`, you don't need to document both the setter and getter. If you generate the doc, this wont render properly

Reply:
OK